### ISSUE TITLE
Setup errors (branch: development_3k)

### ISSUE THREAD

**Issue by ajvazquez** [satnet-project/server#22]
After runing:

     /satnet/server/scripts/server_setup -a

Output:

     [...]

     Copying files to the temporary directory...OK

     Stripping ELF binaries and libraries...OK

     Compressing man pages...OK

     Building file list...OK

     Building Debian package... FAILED!

     *** Failed to build the package

     Do you want to see the log file?  [y]: 

     dpkg-deb: error: parsing file '/var/tmp/tmp.mZjtAke0Rf/package/DEBIAN/control' near line 7 package 'node':
      error in Version string 'v0.12.1-1': version number does not start with digit
     /var/tmp/tmp.mZjtAke0Rf/dpkgbuild.log (END)

     Erasing temporary files...OK

     Writing backup package...OK
     OK

     Deleting temp dir...OK

     dpkg: error processing node_* (--install):
      cannot access archive: No such file or directory
     Errors were encountered while processing:
      node_*
     sudo: npm: command not found
     ./server-setup.sh: line 416: cd: /home/aj/satnet/server/website/static/scripts/: No such file or directory
     ./server-setup.sh: line 417: npm: command not found
     ./server-setup.sh: line 418: bower: command not found
     Press any key to continue

**Comment by ajvazquez** [comment]
Just got the same error with the master branch, so I may be missing some of the dependencies...

**Comment by rtp-calpoly** [comment]
@ajvazquez : this bug is related with the configuration of the node packet. When you are asked in the menu just before this error about the configuration, you should erase the "v" from the version number "v0.12.1-1" so that it looks like "0.12.1-1". This way this problem disappears.

This is one of the things that cannot be automated since the installer of the node package has to be configured through its own user interactive menu. In Debian Wheezy, the node package has to be installed this way, maybe we could check first whether the new Debian version (testing) already includes the node package within the official repositories and, therefore, avoid this step.

Let me know what you think.

**Comment by ajvazquez** [comment]
Thank you! That worked, thanks for all the information.

Later I got another error "No such directory" related to line 386 on the same script. That was solved just by creating the folder $webservices_dir/website/static/scripts/ manually and re-running the script.

Now the server seems to be running, but I get an error when trying to access through the web browser. This is the content of /website/logs/log after the access:

     [Fri Apr 03 22:33:19 2015] [error] >>> Loading local server information...
     [Fri Apr 03 22:33:19 2015] [error] DEBUG (0.001) SELECT "network_server"."id", "network_server"."is_me", "network_server"."is_external", "network_server"."owner_id", "network_server"."identifier", "network_server"."ip_address", "network_server"."latitude", "network_server"."longitude", "network_server"."timestamp" FROM "network_server" WHERE "network_server"."is_me" = true ; args=(True,)
     [Fri Apr 03 22:33:19 2015] [error] [client ::1] mod_wsgi (pid=13745): Target WSGI script '/home/aj/satnet_main/server/WebServices/website/wsgi.py' cannot be loaded as Python module.
     [Fri Apr 03 22:33:19 2015] [error] [client ::1] mod_wsgi (pid=13745): Exception occurred processing WSGI script '/home/aj/satnet_main/server/WebServices/website/wsgi.py'.
     [Fri Apr 03 22:33:19 2015] [error] [client ::1] Traceback (most recent call last):
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/website/wsgi.py", line 47, in <module>
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     server_models.ServerManager().load_local_server()
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/services/network/models/server.py", line 45, in load_local_server
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     s_local = self.get_local()
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/services/network/models/server.py", line 66, in get_local
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     return Server.objects.get(is_me=True)
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/.venv/lib/python2.7/site-packages/django/db/models/manager.py", line 151, in get
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     return self.get_queryset().get(*args, **kwargs)
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/.venv/lib/python2.7/site-packages/django/db/models/query.py", line 304, in get
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     num = len(clone)
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/.venv/lib/python2.7/site-packages/django/db/models/query.py", line 77, in __len__
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     self._fetch_all()
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/.venv/lib/python2.7/site-packages/django/db/models/query.py", line 857, in _fetch_all
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     self._result_cache = list(self.iterator())
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/.venv/lib/python2.7/site-packages/django/db/models/query.py", line 220, in iterator
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     for row in compiler.results_iter():
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/.venv/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 713, in results_iter
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     for rows in self.execute_sql(MULTI):
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/.venv/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 786, in execute_sql
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     cursor.execute(sql, params)
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/.venv/lib/python2.7/site-packages/django/db/backends/util.py", line 69, in execute
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     return super(CursorDebugWrapper, self).execute(sql, params)
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/.venv/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     return self.cursor.execute(sql, params)
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/.venv/lib/python2.7/site-packages/django/db/utils.py", line 99, in __exit__
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     six.reraise(dj_exc_type, dj_exc_value, traceback)
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]   File "/home/aj/satnet_main/server/WebServices/.venv/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]     return self.cursor.execute(sql, params)
     [Fri Apr 03 22:33:19 2015] [error] [client ::1] ProgrammingError: relation "network_server" does not exist
     [Fri Apr 03 22:33:19 2015] [error] [client ::1] LINE 1: ...r"."longitude", "network_server"."timestamp" FROM "network_s...
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]                                                              ^
     [Fri Apr 03 22:33:19 2015] [error] [client ::1]
     
After configuring the apache server with the -x option in the script I get:

     [...]
     ERROR: Site satnet_tls does not exist!
     Site satnet_tls.conf already enabled
     [...]

Although this may not be a problem. Any idea? (This is with the branch "main"). Thank you!

**Comment by rtp-calpoly** [comment]
@ajvazquez : could we separate this new problem in a new issue? This way, we can close this one and open a new issue (better organization)... thanks!

**Comment by rtp-calpoly** [comment]
@ajvazquez : concerning this issue, I have just checked in one of the computers that I have at home that NodeJS and NPM are now part of the official repositories of Debian for the testing release (Jessy). I have already realized that the AngularJS interface has been moved to a separate repository, therefore this issue should never apply again.

The correct answer for this bug is to delete the installation of the AngularJS dependencies (Node, NPM and Bower packages), since now that is all handled from within the "satnet-ng" repository and should never appear again as part of the server installation.
